### PR TITLE
Add `--python` support to `uv run`

### PR DESCRIPTION
**Pull request by zanieb**

### CODE REVIEW

File: crates/uv/src/cli.rs [hunk 1]
```diff
@@ -1654,6 +1654,20 @@ pub(crate) struct RunArgs {
...
+    /// By default, `uv` uses the virtual environment in the current working directory or any parent
+    /// directory, falling back to searching for a Python executable in `PATH`. The `--python`
+    /// option allows you to specify a different interpreter.
+    ///
+    /// Supported formats:
+    /// - `3.10` looks for an installed Python 3.10 using `py --list-paths` on Windows, or
+    ///   `python3.10` on Linux and macOS.
+    /// - `python3.10` or `python.exe` looks for a binary with the given name in `PATH`.
+    /// - `/home/ferris/.local/bin/python3.10` uses the exact Python at the given path.
+    #[arg(long, verbatim_doc_comment, group = "discovery")]
```
Comment (konstin) [inline crates/uv/src/cli.rs:1668]: Do we want `-p` too`

**Review by konstin — APPROVED** [review]

File: crates/uv/src/cli.rs (hunk 1 above)
Comment (zanieb) [reply crates/uv/src/cli.rs:1668]: Probably?

Comment (konstin) [reply crates/uv/src/cli.rs:1668]: I want it :D

**Comment by zanieb** [comment]
```
❯ cargo run -- run -p 3.11 --no-workspace -- python --version
    Finished dev [unoptimized + debuginfo] target(s) in 0.29s
     Running `target/debug/uv run -p 3.11 --no-workspace -- python --version`
Resolved 0 packages in 5ms
Audited 0 packages in 0.38ms
Python 3.11.7
```

**Pull request merged**